packages/jupyter-ai-magics/jupyter_ai_magics/providers.py

```diff
@@ -200,10 +210,39 @@ class ChatOpenAINewProvider(BaseProvider, ChatOpenAI):
     pypi_package_deps = ["openai"]
     auth_strategy = EnvAuthStrategy(name="OPENAI_API_KEY")
 
+class DefaultContentHandler(ContentHandlerBase):
+    content_type = "application/json"
+    accepts = "application/json"
+
+    def transform_input(self, prompt: str, model_kwargs: Dict) -> bytes:
+        input_str = json.dumps({prompt: prompt, **model_kwargs})
+        return input_str.encode('utf-8')
+    
+    def transform_output(self, output: bytes) -> str:
+        response_json = json.loads(output.read().decode("utf-8"))
+        return response_json[0]["generated_text"]
+
 class SmEndpointProvider(BaseProvider, SagemakerEndpoint):
     id = "sagemaker-endpoint"
     name = "Sagemaker Endpoint"
     models = ["*"]
-    model_id_key = "endpoint_name"
     pypi_package_deps = ["boto3"]
     auth_strategy = AwsAuthStrategy()
+
+    content_handler: ContentHandlerBase = DefaultContentHandler
```

Comment:
```suggestion
    content_handler: ContentHandlerBase = DefaultContentHandler()
```

Separately, https://github.com/hwchase17/langchain/pull/3249 has changed the typing for `SagemakerEndpoint` allowed `content_handler`. When we update langchain dep we'll need to update this. 

I've made a PR to update the example: https://github.com/hwchase17/langchain/pull/4598